src/Microsoft.OData.Core/UriParser/Binders/FunctionCallBinder.cs

```diff
@@ -140,16 +140,22 @@ internal static FunctionSignatureWithReturnType MatchSignatureToBuiltInFunction(
 
         /// <summary>
         /// Finds all signatures for the given function name.
+        /// Search in BuiltIn uri functions first, and then in custom uri functions.
         /// </summary>
         /// <param name="functionName">The function to get the signatures for.</param>
         /// <returns>The signatures which match the supplied function name.</returns>
-        internal static FunctionSignatureWithReturnType[] GetBuiltInFunctionSignatures(string functionName)
+        internal static FunctionSignatureWithReturnType[] GetUriFunctionSignatures(string functionName)
         {
-            // Try to find the function in our built-in functions
             FunctionSignatureWithReturnType[] signatures;
-            if (!BuiltInFunctions.TryGetBuiltInFunction(functionName, out signatures))
+
+            // Try to find the function in the user custom functions
+            if (!CustomUriFunctions.TryGetCustomFunction(functionName, out signatures))

```

Comment:
So you say that we keep the 'AddCustomUriFunction' method as is, and modify the 'GetUriFunctionSignatures' method to combine the overloads of both Built-In functions and CustomFunctions. Seems like a better solution : )

Confirm that this is what you ment and I will code, document and add tests.